projects/core/src/window/window-ref.ts

```diff
@@ -42,6 +45,30 @@ export class WindowRef {
     return this.isBrowser() ? window : undefined;
   }
 
+  /**
+   * Returns a reference to the document contained in the window or mimics the document when
+   * the process runs outside the browser.
+   *
+   * The later is useful, when the process runs in SSR and the document is lacking location properties
+   * such as href and origin. These are constructed by leveraging the _injected_ `SERVER_REQUEST_URL`
+   * and `SERVER_REQUEST_ORIGIN`.
+   *
+   * Please be aware that in SSR the document is still lacking the full fletched document object.
+   */
+  get document(): Document {
+    if (!this.isBrowser()) {
+      return {
+        ...this._document,
+        location: {
+          href: this.serverRequestUrl,
+          origin: this.serverRequestOrigin,
+        },
+      } as Document;
+    } else {
+      return this._document;
```

Comment:
[medium] SSR doesn't work for me, complaining:
-  `ERROR TypeError: this.winRef.document.getElementsByTagName is not a function`
 for line
 ```
             const head = this.winRef.document.getElementsByTagName('head')[0];
  ```
- `ERROR TypeError: this.winRef.document.getElementById is not a function`
   ```
           let scriptElement = (this.winRef.document.getElementById(id));
   ```
and couple of other places